nvflare/apis/utils/job_utils.py

```diff
@@ -51,26 +53,33 @@ def convert_legacy_zipped_app_to_job(zip_data: bytes) -> bytes:
     with ZipFile(reader, "r") as in_zip:
         info_list = in_zip.infolist()
         folder_name = info_list[0].filename.split("/")[0]
-        meta_path = normpath_for_zip(os.path.join(folder_name, JobConstants.META_FILE))
-        if next((info for info in info_list if info.filename == meta_path), None):
-            # Already in job layout
-            meta_data = in_zip.read(meta_path)
-            meta = json.loads(meta_data)
-            if JobMetaKey.JOB_FOLDER_NAME.value not in meta:
-                meta[JobMetaKey.JOB_FOLDER_NAME.value] = folder_name
-            else:
-                return zip_data
+        meta_file = os.path.join(folder_name, JobConstants.META)
+        meta_json = os.path.join(folder_name, JobConstants.META_FILE)
```

Comment:
maybe we need 
``meta_json = normpath_for_zip(os.path.join(folder_name, JobConstants.META_FILE))``?